### ISSUE TITLE
Add Void Linux support in Salt

### ISSUE THREAD

**Issue by Vaelatern** [saltstack/salt#30195]
I will be adding support to Salt for my Favorite Linux Distro of the day, [Void Linux](http://www.voidlinux.eu/).
This issue is a place to help consolidate what must be done and try to prevent duplication of effort.
@yopito has also expressed interest in this goal. Please confirm with me how much you have so we can start PRs and I can work to fill in the gaps. If you don't have the time to finish a PR, share diffs with what you have and I'll finish them up to the best of my ability.

**Comment by thatch45** [comment]
++ Void Linux is cool

**Comment by Vaelatern** [comment]
- [x] Service files for Void Linux
- [x] A salt package in Void Linux.
- [ ] xbps pkg module.
- [ ] runit support.
- [ ] Correctly identify a "Void" machine. (I think I know what to do for this, if it isn't already done)

**Comment by Vaelatern** [comment]
@thatch45 I am waiting on @yopito before I start PRing. Once he tells me what he has, and either shares or PRs it, I'll start PRing everything I can think of. More suggestions as to things to check would be welcome.

**Comment by thatch45** [comment]
Sounds good, thanks for the work here!

**Comment by yopito** [comment]
@thatch45 : thanks for your welcome !

this SaltStack PR sounds a little bit early for inclusion in SaltStack since initial support of XBPS and runit/sv is not even finished/fully debugged.

1st step: add XBPS and runit/sv support into the VoidLinux distribution packaging of Salt in in progress.
more details here: https://github.com/voidlinux/void-packages/pull/3326

**Comment by Vaelatern** [comment]
@yopito, Why do you feel we should start pushing patches to the Void Linux package version of salt? Why not start in this repo?

**Comment by yopito** [comment]
@Vaelatern : 
Pushing first to VoidLinux provides a quick way to perform "unit" tests and benefit from VoidLinux users's experience.
if pushed directly to SaltStack, have to wait for its inclusion in an official release before becoming available into VoidLinux (packaging), but we will have to test code against runit/sv and xbps first.

**Comment by Vaelatern** [comment]
@yopito: Accepted. 
I will start testing code you push and sign off your work on the Void Linux repos. After any code is included in Void, it should be PRed to here.

**Comment by yopito** [comment]
hello and sorry for both posting in #31262 and here,
Is it planned to add VoidLinux support into 2016.3 release ? #31262 has been merged into develop branch, but rc2 release does not contain it.
(https://pypi.python.org/packages/source/s/salt/salt-2016.3.0rc2.tar.gz)

**Comment by yopito** [comment]
This issue can be closed: VoidLinux provided support is now included in SaltStack release 2016.11 (thanks !)

**Comment by thatch45** [comment]
Thanks @yopito , and I am glad we could get this in!

**Comment by yopito** [comment]
thanks for your kind words. another PR against `develop` branch is needed too ?

**Comment by thatch45** [comment]
no, we will merge your changes to 2016.11 forward to develop, so you are covered!
